### PR TITLE
Bug/79431 february dates fix

### PR DESCRIPTION
**Pull request by tapiwamukwewa**
#### Checklist

- [ ] Includes tests
- [ ] Update documentation

#### Reviewers should focus on:

Added a validation check for february dates above the 29th (30 & 31)

### CODE REVIEW

File: packages/forms/src/elements/date/date.tsx [hunk 1]
```diff
@@ -18,10 +18,24 @@ const handleChange = (onChange: Function, value: number) => ({
 	}
 };
 
+// JS new Date(2020,02,31) evaluates to a valid date in March, this check is to validate that we still
```
Comment (dotcom9) [inline packages/forms/src/elements/date/date.tsx:21]: This function is a good candidate for a unit test. Perhaps worth adding one?

**Review by dotcom9 — CHANGES_REQUESTED** [review]

**Comment by vrojastpr** [comment]
Agree with the need of unit tests for the `datePassesFebruaryCheck` function.

File: packages/forms/src/__tests__/SameMonthDateValidator.spec.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+let validator = new SameMonthDateValidator();
+
+describe('Date input', ()=>{
+	test('date field is invalid for days 30 and 31 of february ', () => {
+		expect(validator.ResolvedDateIsInSameMonth('2020', '02', '30')).toBeFalsy();
+		expect(validator.ResolvedDateIsInSameMonth('2020', '02', '31')).toBeFalsy();
+	});
+
+	test('date field is valid for day 31 for other months', () => {
+		expect(validator.ResolvedDateIsInSameMonth('2020', '01', '30')).toBeTruthy();
```
Comment (dotcom9) [inline packages/forms/src/__tests__/SameMonthDateValidator.spec.ts:12]: Did you mean this test to be for the 31st Jan?

**Review by dotcom9 — COMMENTED** [review]
Like the unit test and the way you created a separate validator class.

**Review by dotcom9 — APPROVED** [review]

**Review by vrojastpr — APPROVED** [review]

**Pull request merged**